examples/http_server/module/src/lib.rs

```diff
@@ -80,28 +176,150 @@ impl CommandHandler for StaticHttpHandler {
-                    .uri("http://www.google.com")
-                    .body(vec![])
-                    .context("Could not build request")?;
-                http_client
-                    .send_request(request, &Label::public_untrusted())
-                    .context("Could not get response")
-            }
             _ => http::response::Builder::new()
                 .status(http::StatusCode::NOT_FOUND)
                 .body("not found".to_string().into_bytes())
                 .context("Could not build response"),
-        }?;
+        };
 
+        let response = match response {
+            Ok(rsp) => rsp,
+            Err(err) => http::response::Builder::new()
+                .status(http::StatusCode::INTERNAL_SERVER_ERROR)
+                .body(err.to_string().into_bytes())
+                .context("Could not build response")?,
+        };
+
+        info!("Sending the response on the invocation channel!");
         let res = invocation.send(&response);
         invocation.close_channels();
         res?;
         Ok(())
     }
 }
+
+/// A handler corresponding to an HTTP server that redirects every request to
+/// `https://localhost:8080` and responds with `OK` (200) if the redirect is successful.
+pub struct RedirectHandler {
+    helper_invocation_sender: Sender<RedirectInvocation>,
+    client_invocation_sender: Sender<HttpInvocation>,
+}
+
+impl oak::WithInit for RedirectHandler {
+    type Init = RedirectHandlerInit;
+
+    fn create(init: Self::Init) -> Self {
+        oak::logger::init(init.log_sender.unwrap(), log::Level::Debug).unwrap();
+        RedirectHandler {
+            helper_invocation_sender: init.helper_invocation_sender.unwrap(),
+            client_invocation_sender: init.client_invocation_sender.unwrap(),
+        }
+    }
+}
+
+impl CommandHandler for RedirectHandler {
+    type Command = Invocation;
+
+    fn handle_command(&mut self, invocation: Invocation) -> anyhow::Result<()> {
+        info!("Redirecting the request.");
+
+        let label = confidentiality_label(tls_endpoint_tag("localhost:8080"));
+
+        // Create a channel for passing HTTP requests to the HTTP Client node.
+        let (request_writer, request_reader) = oak::io::channel_create("HTTP request", &label)?;
+        // Create a channel for collecting HTTP responses from the HTTP Client node.
+        let (response_writer, response_reader) = oak::io::channel_create("HTTP response", &label)?;
+
+        // Send the request-receiver channel, together with the new invocation source for
+        // communication with the HTTP client pseudo-Node, to the RedirectHelper node in a
+        // RedirectInvocation.
+        let helper_invocation = RedirectInvocation {
+            request_receiver: Some(invocation.clone().receiver.unwrap()),
+            http_invocation_source: Some(HttpInvocationSource {
+                sender: Some(request_writer),
+                receiver: Some(response_reader),
+            }),
+        };
```

Comment:
This pattern is what I was talking about in another comment: now we could have a single method (with 2 generic type params) that creates an invocation (2 channels) and returns both "sides" of the invocation (the invocation itself and the "source"). Perhaps a refactoring for another PR, but could you create an issue to track?

Reply:
Created #1832.